### PR TITLE
[Enhancement] (nereids)implement showRolesCommand in nereids

### LINKED ISSUE #42768: [Enhancement] (nereids)implement showRolesCommand in nereids
showRoles command is supported in legacy planner but not in nereids(new doris fe planner). you need implement the following showRoles command in nereids:
unsupportedshowRolesStatement
``` SHOW ROLES                                                                    ```

### PR DESCRIPTION
**Pull request by Vallishp**
### What problem does this PR solve?

Support show roles in nereids.

Issue Number: close #42768 

Related PR: #xxx

Problem Summary:

### Check List (For Committer)

- Test 

    - [*] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No colde files have been changed.
        - [ ] Other reason 
2024-11-08 03:37:22.042 INFO [suite-thread-1] (test_nereids_role.groovy:64) - roles_drop: [[admin, null, 'admin'@'%', Admin_priv, null, null, null, null, null, null, null, null, null], [operator, null, 'root'@'%', Node_priv,Admin_priv, null, null, null, null, null, null, null, null, null]]
2024-11-08 03:37:22.100 INFO [suite-thread-1] (ScriptContext.groovy:120) - Run test_nereids_role in /root/doris/workspace/doris/regression-test/suites/nereids_p0/ddl/account/test_nereids_role.groovy succeed
2024-11-08 03:37:52.786 INFO [main] (RegressionTest.groovy:281) - Start to run single scripts
2024-11-08 03:39:04.628 INFO [main] (RegressionTest.groovy:410) - Success suites:
/root/doris/workspace/doris/regression-test/suites/nereids_p0/ddl/account/test_nereids_role.groovy: group=default,p0, name=test_nereids_role
2024-11-08 03:39:04.629 INFO [main] (RegressionTest.groovy:492) - All suites success.
 ____   _    ____ ____  _____ ____
|  _ \ / \  / ___/ ___|| ____|  _ \
| |_) / _ \ \___ \___ \|  _| | | | |
|  __/ ___ \ ___) |__) | |___| |_| |
|_| /_/   \_\____/____/|_____|____/

2024-11-08 03:39:04.629 INFO [main] (RegressionTest.groovy:440) - Test 1 suites, failed 0 suites, fatal 0 scripts, skipped 0 scripts
2024-11-08 03:39:04.629 INFO [main] (RegressionTest.groovy:126) - Test finished

- Behavior changed:

    - [ ] No. No change.
    - [ ] Yes. 

- Does this need documentation?

    - [ ] No.
    - [ ] Yes. 

- Release note

    
    
    None

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by Vallishp** [comment]
run buildall

**Review by LiBinfeng-01 — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowRolesCommand.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    @Override
+    public void run(ConnectContext ctx, StmtExecutor executor) throws Exception {
+        if (!Env.getCurrentEnv().getAccessManager().checkGlobalPriv(ConnectContext.get(), PrivPredicate.GRANT)) {
+            ErrorReport.reportAnalysisException(ErrorCode.ERR_SPECIFIC_ACCESS_DENIED_ERROR, "GRANT");
+        }
+
+        List<List<String>> infos = Env.getCurrentEnv().getAuth().getRoleInfo();
+        ShowResultSet resultSet = new ShowResultSet(getMetaData(), infos);
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowRolesCommand.java:85]: remove getMetaData() method, use META_DATA directly

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowRolesCommand.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+
+/**
+ * show roles command
+ */
+public class ShowRolesCommand extends Command implements NoForward {
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowRolesCommand.java:42]: The show command related features have some common logical need to be sorted out. I am working on the common part and the pr is https://github.com/apache/doris/pull/43271 . You can take it as a reference, the most important part is all show commands should inherent from ShowCommand and override the doRun method to do the real job.(you can move the code in run() function to doRun())  Sorry for the late change, you might have to wait the pr is merged and update all other show command prs accordingly.

Comment (Vallishp) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowRolesCommand.java:42]: Ok. sure thanks.  will handle all show related PR after merge of #43271

**Comment by Vallishp** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowRolesCommand.java (hunk 2 above)
Comment (Vallishp) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowRolesCommand.java]: @starocean999  i have updated as per your comments. Please help to review again.  after that i will update all other show command PRs accordingly. thank you

**Comment by Vallishp** [comment]
run buildall

**Comment by Vallishp** [comment]
run p0

**Review by starocean999 — APPROVED** [review]
LGTM

**Pull request merged**